lib/msf/core/module/data_store.rb

```diff
@@ -34,7 +39,7 @@ def import_defaults(clear_datastore = true)
   def import_target_defaults
     return unless defined?(targets) && targets && target && target.default_options
 
-    datastore.import_options_from_hash(target.default_options, true, 'self')
+    datastore.import_defaults_from_hash(target.default_options, imported_by: 'self')
```

Comment:
There's an existing bug here that I may or may not fix as part of this pull request:

Changing targets leaves state behind in the datastore  - https://github.com/rapid7/metasploit-framework/issues/16932